lib/ibmdb.js

```diff
@@ -190,73 +189,79 @@ IBMDB.prototype.escapeName = function(name) {
-    if (res) {
-      offset = Number(res[1]);
-      sql = sql.replace(self.offsetRE, '');
+  // var executeStatement = function(conn, sql, params, options, callback) {
+  var executeStatement = function(conn) {
+    // var self = this;
+    var limit = 0;
+    var offset = 0;
+    var stmt = {};
+
+    stmt.noResults = (options && options.noResults) ? options.noResults : false;
+    // This is standard DB2 syntax. LIMIT and OFFSET
+    // are configured off by default. Enable these to
+    // leverage LIMIT and OFFSET.
+    if (!self.useLimitOffset) {
+      var res = sql.match(self.limitRE);
+      if (res) {
+        limit = parseInt(res[1], 10);
+        sql = sql.replace(self.limitRE, '');
+      }
+      res = sql.match(self.offsetRE);
+      if (res) {
+        offset = parseInt(res[1], 10);
+        sql = sql.replace(self.offsetRE, '');
+      }
     }
-  }
 
-  // Build the stmt object that will be passed into the query call.
-  // This is done because the query call can take an object or a set
-  // of parameters.  Depending on the SQL being passed in the call with
-  // parameters may fail due to improper handling in the ibm_db module.
-  stmt.sql = sql;
-  stmt.params = params;
-
-  conn.query(stmt, function(err, data, more) {
-    debug('DB2.prototype.executeSQL (exit)' +
-          ' stmt=%j params=%j err=%j data=%j more=%j',
-          stmt, err, data, more);
-    // schedule callback as there is more code to
-    // execute in the db2 driver to cleanup the current query
-    if (offset || limit) {
-      data = data.slice(offset, offset + limit);
-    }
+    // Build the stmt object that will be passed into the query call.
+    // This is done because the query call can take an object or a set
+    // of parameters.  Depending on the SQL being passed in the call with
+    // parameters may fail due to improper handling in the ibm_db module.
+    stmt.sql = sql;
+    stmt.params = params;
+
+    conn.query(stmt, function(err, data, more) {
+      debug('DB2.prototype.executeSQL (exit)' +
+            ' stmt=%j params=%j err=%j data=%j more=%j',
+            stmt, params, err, data, more);
+      // schedule callback as there is more code to
+      // execute in the db2 driver to cleanup the current query
+      if (offset || limit) {
+        data = data.slice(offset, offset + limit);
+      }
 
-    if (!err) {
-      if (more) {
-        process.nextTick(function() {
-          return callback(err, data);
-        });
+      if (!err) {
+        if (more) {
+          process.nextTick(function() {
+            return callback(err, data);
+          });
+        }
       }
-    }
 
-    callback && callback(err, data);
-  });
+      callback && callback(err, data);
+    });
+  };
+
+  if (options.transaction) {
+    var conn = options.transaction.connection;
+    executeStatement(conn);
+  } else {
+    this.connect(function(err, conn) {
+      if (err) return callback(err);
+
+      executeStatement(conn);
+    });
+  }
```

Comment:
I don't agree with putting the func under (only in test cases. In this case, it makes more sense declearing the function first and then using it. However, you should declare the function like:

```
function executeStatement(conn) {
}
```

instead of `var executeStatement...`